packages/number-field/src/vaadin-number-field.js

```diff
@@ -163,12 +161,14 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
        */
       step: {
         type: Number,
-        value: 1,
-        observer: '_stepChanged',
       },
     };
   }
 
+  static get observers() {
+    return ['_minChanged(min, inputElement)', '_maxChanged(max, inputElement)', '_stepChanged(step, inputElement)'];
```

Comment:
Looks like there is no specific logic for `min` and `max`, so you could consider using the mixin:

```js
  static get delegateProps() {
    return [...super.delegateProps, 'min', 'max'];
  }
```